### PR TITLE
test(core): added tests with synchronous executor for async methods

### PR DESCRIPTION
**Pull request by Johaney-s**
* Asynchronous methods have to be performed synchronously to run tests
* Added tests for methods dealing with autoassigning subgroups

### CODE REVIEW

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java [hunk 1]
```diff
@@ -5386,6 +5386,88 @@ public void getTotalGroupStatusForMembers_notMembersOfGroup() throws Exception {
...
+		perun.getGroupsManager().createGroup(sess, vo, group);
+		perun.getGroupsManager().createGroup(sess, group, group2);
+		perun.getGroupsManager().createGroup(sess, vo, group3);
+
+		perun.getResourcesManagerBl().assignGroupToResource(sess, group3, resource, false, false, true);
+		assertThat(perun.getResourcesManagerBl().getAssignedResources(sess, group2).isEmpty());
+
+		// moving to autoassigning tree, should get autoassigned
+		groupsManagerBl.moveGroup(sess , group3, group2);
+		assertTrue(perun.getResourcesManagerBl().getAssignedResources(sess, group3).size() == 1);
```
Comment (cuadradek) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java:5448]:
```suggestion
		assertTrue(perun.getResourcesManagerBl().getAssignedResources(sess, group2).size() == 1);
```

**Review by cuadradek — CHANGES_REQUESTED** [review]
Great work!

**Review by cuadradek — APPROVED** [review]

**Review by metodej — CHANGES_REQUESTED** [review]
Code looks great, but could please you add tests of method `activateGroupResourceAssignment` in async variant as well?

**Comment by Johaney-s** [comment]
> Code looks great, but could please you add tests of method `activateGroupResourceAssignment` in async variant as well?

Thanks, added `activateGroupResourceAssignmentAsync` test.

**Review by metodej — APPROVED** [review]

**Pull request merged**